cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -144,10 +152,6 @@ public synchronized void create(final NamespaceMeta metadata) throws Exception {
       authorizerInstantiator.get().grant(namespace, principal, ImmutableSet.of(Action.ALL));
     }
 
-    // store the meta first in the namespace store because namespacedlocationfactory need to look up location
-    // mapping from namespace config
-    nsStore.create(metadata);
```

Comment:
We still use it in AbstractStorageProviderNamespaceAdmin#create.
If exploreFacade#createNamespace fails, we call deleteLocation, which does a lookup on the nsStore.